app/api/__init__.py

```diff
@@ -15,7 +16,14 @@
 
 # users
 api.route(UserList, 'user_list', '/users')
-api.route(UserDetail, 'user_detail', '/users/<int:id>')
+api.route(UserDetail, 'user_detail', '/users/<int:id>', '/notifications/<int:notification_id>/user')
+api.route(UserRelationship, 'user_notification', '/users/<int:id>/relationships/notifications')
+
+#notifications
+api.route(NotificationList, 'notification_list', '/notifications', '/users/<int:id>/notifications')
+api.route(NotificationDetail, 'notification_detail', '/notifications/<int:id>', 'users/<int:id>/notifications')
```

Comment:
I didn't get the end point here ```users/<int:id>/notifications``` You are asking for Notification detail and yet the end point is to get all notifications of user.
I think it should be ```users/<int:user_id>/notifications/<int:id>``` but then it doesn't make much sense and it is same as ``` /notifications/<int:id>```.  @shubham-padia @niranjan94 @SaptakS Can you please confirm.